syntax = "proto2";

enum Alignment {
  Ally = 0;
  Enemy = 1;
}

enum Class {
  Corvette = 0;
  Dreadnought = 1;
}

message Officer {
  required string first_name = 1;
  required string last_name = 2;
  required string rank = 3;
}

message Spaceship {
  required Alignment alignment = 1;
  required string name = 2;
  required Class classification = 3;
  required float length = 4;
  required int32 crew_size = 5;
  required bool armed = 6;
  repeated Officer officers = 7;
}

message Coordinates {
  required string coordinates = 1;
}

service Reporting {
  rpc GetSpaceships (Coordinates) returns (stream Spaceship);
}
